File: GTUC Voting System/LoginForm.vb
Imports FireSharp.Config
Imports FireSharp.Response
Imports FireSharp.Interfaces



Public Class LoginForm
    Private fcon As New FirebaseConfig() With
      {
    .AuthSecret = "YSv44ecN1DUdKnvTWf9qn8iMuxcNLScFC5N2emRM",
    .BasePath = "https://vbeng-25e59.firebaseio.com/"
      }

    Private client As IFirebaseClient


    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Try     'Setting up Database Connection
            client = New FireSharp.FirebaseClient(fcon)
        Catch ex As Exception
            MessageBox.Show("There was a problem with the connection")
        End Try
    End Sub


    Private Sub loginBtn_Click(sender As Object, e As EventArgs) Handles loginBtn.Click
        'Validate Login Input
        Dim res = client.Get("StudentList/" + indexText.Text)
        Dim User As New Student()
        User = res.ResultAs(Of Student)() 'Getting Database result

        If indexText.Text = Nothing Or PasswordText.Text = Nothing Then
            MsgBox("Please Fill All The Fields", MsgBoxStyle.Exclamation)
            indexText.Clear()
            PasswordText.Clear()
            indexText.Focus()
        ElseIf indexText.Text <> User.IndexNO Or PasswordText.Text <> User.Password Then
            MsgBox("Email or Password Is Incorrect", MsgBoxStyle.Exclamation)
            indexText.Clear()
            PasswordText.Clear()
            indexText.Focus()
        Else
            Vote.Show()


        End If
    End Sub

    Private Sub usernameText_TextChanged(sender As Object, e As EventArgs) Handles indexText.TextChanged

    End Sub

    Private Sub PasswordText_TextChanged(sender As Object, e As EventArgs) Handles PasswordText.TextChanged

    End Sub
End Class

File: login/LoginForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using FireSharp.Config;
using FireSharp.Response;
using FireSharp.Interfaces;



namespace login
{
    public partial class LoginForm : Form
    {
        public LoginForm()
        {
            InitializeComponent();
        }


        IFirebaseConfig ifc = new FirebaseConfig()
        {
            AuthSecret = "YSv44ecN1DUdKnvTWf9qn8iMuxcNLScFC5N2emRM",
            BasePath = "https://vbeng-25e59.firebaseio.com/"
        };

        IFirebaseClient client;

        private void LoginForm_Load(object sender, EventArgs e)
        {
            {
                try
                {
                    client = new FireSharp.FirebaseClient(ifc);
                }

                catch
                {
                    MessageBox.Show("No Internet or Connection Problem");
                }
            }
        }

 
        private void LGbtn_Click(object sender, EventArgs e)
        {
            #region Condition
            if (string.IsNullOrWhiteSpace(index.Text) &&
               string.IsNullOrWhiteSpace(password.Text))
            {
                MessageBox.Show("Please Fill All The Fields");
                return;
            }
            #endregion

            FirebaseResponse res = client.Get(@"StudentList/" + index.Text);
            User ResUser = res.ResultAs<User>();// database result

            User CurUser = new User() // USER GIVEN INFO
            { 
                IndexNo = index.Text,
                Password = password.Text
            };

            if (User.IsEqual(ResUser, CurUser))
            {
                MessageBox.Show("Welcome");

            }

            else
            {
                User.ShowError();
            }
        }
    }
}